ci-operator/step-registry/aws/provision/dns-firewall/aws-provision-dns-firewall-commands.sh

```diff
@@ -0,0 +1,186 @@
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+REGION=${REGION:-$LEASED_RESOURCE}
+
+# Domain list
+OCP_DOMAIN_ALLOW_LIST=$(echo -e '
+  "registry.redhat.io",
+  "access.redhat.com",
+  "quay.io",
+  "sso.redhat.com",
+
+  "cert-api.access.redhat.com",
+  "api.access.redhat.com",
+  "infogw.api.openshift.com",
+  "console.redhat.com",
+
+  "ec2.amazonaws.com",
+  "events.amazonaws.com",
+  "iam.amazonaws.com",
+  "route53.amazonaws.com",
+  "s3.amazonaws.com",
+  "s3.'${REGION}'.amazonaws.com",
+  "s3.dualstack.'${REGION}'.amazonaws.com",
+  "sts.amazonaws.com",
+  "sts.'${REGION}'.amazonaws.com",
+  "tagging.us-east-1.amazonaws.com",
+  "ec2.'${REGION}'.amazonaws.com",
+  "elasticloadbalancing.'${REGION}'.amazonaws.com",
+  "servicequotas.'${REGION}'.amazonaws.com",
+  "tagging.'${REGION}'.amazonaws.com",
+
+  "mirror.openshift.com",
+  "storage.googleapis.com",
+  "quayio-production-s3.s3.amazonaws.com",
+  "api.openshift.com",
+  "rhcos.mirror.openshift.com"
+')
+
+OSD_DOMAIN_ALLOW_LIST=$(echo -e '
+  "registry.redhat.io",
+  "quay.io",
+  "sso.redhat.com",
+  "quay-registry.s3.amazonaws.com",
+  "ocm-quay-production-s3.s3.amazonaws.com",
+  "quayio-production-s3.s3.amazonaws.com",
+  "cart-rhcos-ci.s3.amazonaws.com",
+  "openshift.org",
+  "registry.access.redhat.com",
+  "registry.connect.redhat.com",
+  "pull.q1w2.quay.rhcloud.com",
+  "q1w2.quay.rhcloud.com",
+  "www.okd.io",
+  "www.redhat.com",
+  "aws.amazon.com",
+  "catalog.redhat.com",
+
+  "cert-api.access.redhat.com",
+  "api.access.redhat.com",
+  "infogw.api.openshift.com",
+  "console.redhat.com",
+  "cloud.redhat.com",
+  "observatorium-mst.api.openshift.com",
+  "observatorium.api.openshift.com",
+
+  "ec2.amazonaws.com",
+  "events.'${REGION}'.amazonaws.com",
+  "iam.amazonaws.com",
+  "route53.amazonaws.com",
+  "sts.amazonaws.com",
+  "sts.'${REGION}'.amazonaws.com",
+  "tagging.us-east-1.amazonaws.com",
+  "ec2.'${REGION}'.amazonaws.com",
+  "elasticloadbalancing.'${REGION}'.amazonaws.com",
+  "servicequotas.'${REGION}'.amazonaws.com",
+  "tagging.'${REGION}'.amazonaws.com",
+
+  "mirror.openshift.com",
+  "storage.googleapis.com",
+  "api.openshift.com",
+
+  "api.pagerduty.com",
+  "events.pagerduty.com",
+  "api.deadmanssnitch.com",
+  "nosnch.in",
+  "osdsecuritylogs.splunkcloud.com",
+  "http-inputs-osdsecuritylogs.splunkcloud.com",
+  "sftp.access.redhat.com"
+')
+
+DOMAIN_LIST=${OCP_DOMAIN_ALLOW_LIST}
+if [[ ${MANAGED_CLUSTER} == "yes" ]]; then
+  DOMAIN_LIST=${OSD_DOMAIN_ALLOW_LIST}
+fi
+
+function aws_add_param_to_json() {
+    local k="$1"
+    local v="$2"
+    local param_json="$3"
+    if [ ! -e "$param_json" ]; then
+        echo -n '[]' > "$param_json"
+    fi
+    cat <<< "$(jq  --arg k "$k" --arg v "$v" '. += [{"ParameterKey":$k, "ParameterValue":$v}]' "$param_json")" > "$param_json"
+}
+
+# Create DNS firewall
+UNIQUE_HASH=$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 4)
```

Comment:
I believe there is no restrictions here, so please don't override the `UNIQUE_HASH`.

```suggestion
```